### PR TITLE
feat: MSPs add value propositions

### PR DESCRIPTION
**Pull request by snowmead**
MSPs must sign up with a single value proposition to start with. 
Afterwards they could call the new `add_value_prop` extrinsic to create new value propositions. If they wish to no longer support a value prop, they can call the new `make_value_prop_unavailable`.

Users are able to choose a value proposition when creating a bucket.

Buckets are unaffected by value props becoming unavailable. They continue to charge based on their value prop. Though buckets can only be created based on available value props.

### CODE REVIEW

**Review by links234 — COMMENTED** [review]
Left some comments and nitpicks. Looks good overall.

File: test/util/bspNet/helpers.ts [hunk 1]
```diff
@@ -256,11 +264,8 @@ export const runSimpleBspNet = async (bspNetConfig: BspNetConfig, verbose = fals
           bspNetConfig.capacity || ShConsts.CAPACITY_512,
           // The peer ID has to be different from the BSP's since the user now attempts to send files to MSPs when new storage requests arrive.
           ["/ip4/127.0.0.1/tcp/30350/p2p/12D3KooWNEZ8PGNydcdXTYy1SPHvkP9mbxdtTqGGFVrhorDzeTfA"],
-          {
-            identifier: ShConsts.VALUE_PROP,
```
Comment (ffarall) [inline test/util/bspNet/helpers.ts:260]: I believe this constant isn't used anymore

**Review by ffarall — APPROVED** [review]
Overall great job! I subscribe to Alex's comments

File: pallets/providers/src/types.rs [hunk 2]
```diff
@@ -8,17 +8,38 @@ use frame_system::pallet_prelude::BlockNumberFor;
...
+pub type Multiaddresses<T> = BoundedVec<MultiAddress<T>, MaxMultiAddressAmount<T>>;
+
 #[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
 #[scale_info(skip_type_params(T))]
 pub struct ValueProposition<T: Config> {
-    pub identifier: ValuePropId<T>,
-    pub data_limit: StorageDataUnit<T>,
-    pub protocols: BoundedVec<Protocols<T>, MaxProtocols<T>>,
-    // todo!("add relevant fields here")
+    pub price_per_unit_of_data_per_block: BalanceOf<T>,
```
[the remark this answers is not in the document]
Comment (snowmead) [reply pallets/providers/src/types.rs:25]: Added `commitment` to value proposition

**Review by links234 — APPROVED** [review]

**Pull request merged**